### ISSUE TITLE
Nextcloud 20.0.5 ships old version of Deck app

### ISSUE THREAD

**Issue by xshadow** [nextcloud/server#25172]
Dear Nextcloud team, when upgrading nextcloud from 20.0.4 to 20.0.5 the Deck App is broken. 

This results in log messages like this: 

```
Jan 17 00:26:48 cloud NextCloud[14718]: {"reqId":"cDXcszbKM0eV92Oo8iPU","level":3,"time":"2021-01-16 23:26:48.096800","remoteAddr":"0:0:0:0:0:0:0:0","user":"user","app":"no app in context","method":"GET","url":"/ocs/v2.php/core/navigation/apps?absolute=true&format=json","message":"{\"Exception\":\"InvalidArgumentException\",\"Message\":\"Notifier Deck (id: deck) is not considered because it is using the old way to register.\",\"Code\":0,\"Trace\":[{\"file\":\"/var/www/cloud.foobar.org/nextcloud-20.0.5/apps/deck/lib/AppInfo/Application.php\",\"line\":122,\"function\":\"registerNotifier\",\"class\":\"OC\\\\Notification\\\\Manager\",\"type\":\"->\",\"args\":[{\"__class__\":\"Closure\"},{\"__class__\":\"Closure\"}]},{\"file\":\"/var/www/cloud.foobar.org/nextcloud-20.0.5/apps/deck/appinfo/app.php\",\"line\":26,\"function\":\"registerNotifications\",\"class\":\"OCA\\\\Deck\\\\AppInfo\\\\Application\",\"type\":\"->\",\"args\":[]},{\"file\":\"/var/www/cloud.foobar.org/nextcloud-20.0.5/lib/private/legacy/OC_App.php\",\"line\":289,\"args\":[\"/var/www/cloud.foobar.org/nextcloud-20.0.5/apps/deck/appinfo/app.php\"],\"function\":\"require_once\"},{\"file\":\"/var/www/cloud.foobar.org/nextcloud-20.0.5/lib/private/legacy/OC_App.php\",\"line\":171,\"function\":\"requireAppFile\",\"class\":\"OC_App\",\"type\":\"0:0:0:0:0:0:0:0\",\"args\":[{\"__class__\":\"OCA\\\\Deck\\\\AppInfo\\\\Application\"}]},{\"file\":\"/var/www/cloud.foobar.org/nextcloud-20.0.5/lib/private/legacy/OC_App.php\",\"line\":131,\"function\":\"loadApp\",\"class\":\"OC_App\",\"type\":\"0:0:0:0:0:0:0:0\",\"args\":[\"deck\"]},{\"file\":\"/var/www/cloud.foobar.org/nextcloud-20.0.5/ocs/v1.php\",\"line\":56,\"function\":\"loadApps\",\"class\":\"OC_App\",\"type\":\"0:0:0:0:0:0:0:0\",\"args\":[]},{\"file\":\"/var/www/cloud.foobar.org/nextcloud-20.0.5/ocs/v2.php\",\"line\":24,\"args\":[\"/var/www/cloud.foobar.org/nextcloud-20.0.5/ocs/v1.php\"],\"function\":\"require_once\"}],\"File\":\"/var/www/cloud.foobar.org/nextcloud-20.0.5/lib/private/Notification/Manager.php\",\"Line\":92,\"CustomMessage\":\"--\"}","userAgent":"Mozilla/5.0 (Macintosh) mirall/3.0.3git (build 3799) (Nextcloud)","version":"0.0.0.0"}
```

Debugging to the root cause, it seems that Nextcloud 20.0.5 ships a fairly old version of the Deck App. 

```
root@cloud:~# cat /var/www/cloud.foobar.org/nextcloud-20.0.5/apps/deck/CHANGELOG.md 
# Changelog
All notable changes to this project will be documented in this file.

## 0.4.1 - 2018-07-28

### Added

- Make app compatible with Nextcloud 14

### Fixed 
```

```
root@cloud:~# cat /var/www/cloud.foobar.org/nextcloud-20.0.4/apps/deck/CHANGELOG.md 
# Changelog
All notable changes to this project will be documented in this file.

## 1.2.3 - 2021-01-04

### Fixed
* [#2622](https://github.com/nextcloud/deck/pull/2622) Fix gradient and stack header spacing for safari
* [#2626](https://github.com/nextcloud/deck/pull/2626) Adding a description icon to cards when they contain a description without any checkmarks @MonkeySon
* [#2659](https://github.com/nextcloud/deck/pull/2659) Matching color of description cursor with text color @JonFStr
* [#2676](https://github.com/nextcloud/deck/pull/2676) Only load filter view when shown
* [#2680](https://github.com/nextcloud/deck/pull/2680) Do not try to add change data if it doesn't exist
* [#2681](https://github.com/nextcloud/deck/pull/2681) Filter out deleted stacks from results
```

It seems that other people are also affected by this: https://help.nextcloud.com/t/deck-app-coudnt-install-latest-version/104522

Thank you for Nextcloud!

**Comment by xshadow** [comment]
I might be wrong, since nextcloud is not shipping any deck app by default.

**Comment by xshadow** [comment]
This happened due to a broken deployment.

It resulted in a migration to version 1.2.3 did not run properly. It could be fixed with putting the decks code of version 1.2.3 into  `/var/www/cloud.foobar.org/nextcloud-20.0.5/apps/deck/`. And then executing the following steps: 

Check the db version: 

```
MariaDB [nextcloud]> select * from migrations where app = 'deck';
+------+-------------------------+
| app  | version                 |
+------+-------------------------+
| deck | 1000Date20200306161713  |
| deck | 1000Date20200308073933  |
| deck | 10200Date20201111150114 |
+------+-------------------------+
3 rows in set (0.001 sec)
```

Delete the incomplete migration:
```sql
use nextcloud;
delete from migrations where app = 'deck' and version = '10200Date20201111150114';
```
Re-run the migration:
```shell
cd /var/www/cloud.foobar.org/nextcloud-20.0.5
php ./occ migrations:execute -vvv deck 10200Date20201111150114
```

Content of the migration: https://github.com/nextcloud/deck/pull/2521/files#diff-afa623449d243e339248c502885b1783d25600e37379abaebc71f8f9a79a68bd
